easybuild/easyblocks/t/tensorflow.py

```diff
@@ -835,41 +838,98 @@ def test_step(self):
             try:
-                num_test_jobs = min(self.cfg['parallel'], max(1, int(gpu_ct.strip()))) if ec == 0 else 1
-                test_opts.append('--test_env=TF_GPU_COUNT=%s' % num_test_jobs)
-                test_opts.append('--test_env=TF_TESTS_PER_GPU=1')
+                if ec != 0:
+                    raise RuntimeError()  # Caught below
+                gpu_ct = max(1, int(gpu_ct.strip()))
             except Exception:
                 self.log.info('Failed to get the number of GPUs on this system. Using only 1 test job')
-                num_test_jobs = 1
-        test_opts.append('--local_test_jobs=%s' % num_test_jobs)
+                gpu_ct = 1
 
-        # Don't build tests which won't be executed
-        test_opts.append('--build_tests_only')
+            num_gpus_to_use = min(self.cfg['parallel'], gpu_ct)
+            # Can (likely) only run 1 test per GPU but don't need to limit CPU tests
+            num_test_jobs = {'cpu': self.cfg['parallel'], 'gpu': num_gpus_to_use}
 
-        test_tag_filters = self.cfg['test_tag_filters']
-        if test_tag_filters:
-            # Add both build and test tag filters as done by the TF CI scripts
-            test_opts.extend("--%s_tag_filters='%s'" % (step, test_tag_filters) for step in ('test', 'build'))
+        # These are used by the `parallel_gpu_execute` helper script from TF
+        test_opts.append('--test_env=TF_GPU_COUNT=%s' % num_test_jobs['gpu'])
+        test_opts.append('--test_env=TF_TESTS_PER_GPU=1')
 
-        # Compose final command
-        cmd = (
-            [self.cfg['pretestopts']]
-            + ['bazel']
-            + self.bazel_opts
-            + ['test']
-            + test_opts
-            + [self.cfg['testopts'], '--']
-            # specify targets to test as last argument
-            + test_targets
-        )
+        cfg_testopts = {'cpu': self.cfg['testopts'], 'gpu': self.cfg['testopts_gpu']}
 
-        run_cmd(' '.join(cmd), log_all=True, simple=True)
+        for device in ('cpu', 'gpu'):
+            # Determine tests to run
+            test_tag_filters_name = 'test_tag_filters_' + device
+            test_tag_filters = self.cfg[test_tag_filters_name]
+            if not test_tag_filters:
+                self.log.info('Skipping %s test because %s is not set', device, test_tag_filters_name)
+                continue
+            else:
+                self.log.info('Starting %s test', device)
+
+            current_test_opts = test_opts[:]
+            current_test_opts.append('--local_test_jobs=%s' % num_test_jobs[device])
+
+            # Add both build and test tag filters as done by the TF CI scripts
+            current_test_opts.extend("--%s_tag_filters='%s'" % (step, test_tag_filters) for step in ('test', 'build'))
+
+            # TF can only run tests explicitely marked as 'gpu' on GPUs and those don't work on cpu
+            # See https://github.com/tensorflow/tensorflow/issues/45664
+            if device == 'cpu':
```

Comment:
Shouldn't this be `gpu`?

If not, please clarify the comment, since this doesn't make sense at first glance...

Reply:
Oh, it's `-1`... 🤦 

OK, please mention what `-1` does in the comment, it's easy to overlook that as `1` which strongly suggests "enable"...

Reply:
Oh, it's `-1`... 🤦 

OK, please mention what `-1` does in the comment, it's easy to overlook that as `1` which strongly suggests "enable"...

Reply:
I changed the comment to say "Disable GPUs..." (or so), is that enough?